File: app/src/main/java/com/stedi/multitouchpaint/dialogs/BaseDialog.java
package com.stedi.multitouchpaint.dialogs;

import android.app.DialogFragment;
import android.app.FragmentManager;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.stedi.multitouchpaint.R;

import butterknife.ButterKnife;
import butterknife.Unbinder;

abstract class BaseDialog extends DialogFragment {
    private Unbinder unbinder;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(DialogFragment.STYLE_NORMAL, R.style.BaseDialog);
    }

    protected View butterKnifeInflate(LayoutInflater inflater, int layoutResId, ViewGroup parent) {
        View root = inflater.inflate(layoutResId, parent, false);
        unbinder = ButterKnife.bind(this, root);
        return root;
    }

    @Override
    public void show(FragmentManager manager, String tag) {
        try {
            super.show(manager, tag);
        } catch (IllegalStateException ex) {
            ex.printStackTrace();
        }
    }

    @Override
    public void dismiss() {
        super.dismissAllowingStateLoss();
    }

    @Override
    public void onDestroyView() {
        if (unbinder != null)
            unbinder.unbind();
        super.onDestroyView();
    }
}

File: app/src/main/java/com/stedi/multitouchpaint/dialogs/BaseDialog.kt
package com.stedi.multitouchpaint.dialogs

import android.app.DialogFragment
import android.app.FragmentManager
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import butterknife.ButterKnife
import butterknife.Unbinder
import com.stedi.multitouchpaint.R

abstract class BaseDialog : DialogFragment() {
    private var unbinder: Unbinder? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(DialogFragment.STYLE_NORMAL, R.style.BaseDialog)
    }

    protected fun butterKnifeInflate(inflater: LayoutInflater, layoutResId: Int, parent: ViewGroup?): View {
        val root = inflater.inflate(layoutResId, parent, false)
        unbinder = ButterKnife.bind(this, root)
        return root
    }

    override fun show(manager: FragmentManager, tag: String) {
        try {
            super.show(manager, tag)
        } catch (e: IllegalStateException) { // it happens...
            e.printStackTrace()
        }
    }

    override fun dismiss() {
        super.dismissAllowingStateLoss()
    }

    override fun onDestroyView() {
        val unbinder = unbinder
        if (unbinder != null) {
            unbinder.unbind()
        }
        super.onDestroyView()
    }
}